apps/spruce/src/pages/version/TestAnalysis/utils.ts

```diff
@@ -0,0 +1,161 @@
+import { TaskStatus } from "@evg-ui/lib/types/task";
+import { Unpacked } from "@evg-ui/lib/types/utils";
+import {
+  TestAnalysisQueryTasks,
+  TaskBuildVariantField,
+  GroupedTestMap,
+} from "./types";
+
+/**
+ * `groupTestsByName` takes an array of tasks and groups them by test name. Preserves the task name, build variant, id, and status of each task.
+ * @param tasks - an array of tasks
+ * @returns - a map of test names to an array of tasks
+ */
+const groupTestsByName = (
+  tasks: TestAnalysisQueryTasks,
+): Map<string, TaskBuildVariantField[]> => {
+  const testMap = new Map<string, TaskBuildVariantField[]>();
+  const processedTasks = new Set<string>();
+
+  tasks.forEach((task) => {
+    const tests = getTestsInTask(task.tests);
```

Comment:
`getTestsInTask` seems like it just returns a subset of the fields in `task.tests.testResults`; should we consider skipping this operation and iterating on `task.tests.testResults` directly?